core/dbt/events/base_types.py

```diff
@@ -61,24 +57,48 @@ class EventLevel(str, Enum):
     ERROR = "error"
 
 
-@dataclass
 class BaseEvent:
     """BaseEvent for proto message generated python events"""
 
-    #   def __post_init__(self):
-    #       super().__post_init__()
-    #       if not self.info.level:
-    #           self.info.level = self.level_tag()
-    #       assert self.info.level in ["info", "warn", "error", "debug", "test"]
-    #       if not hasattr(self.info, "msg") or not self.info.msg:
-    #           self.info.msg = self.message()
-    #       self.info.invocation_id = get_invocation_id()
-    #       self.info.extra = get_global_metadata_vars()
-    #       self.info.ts = datetime.utcnow()
-    #       self.info.pid = get_pid()
-    #       self.info.thread = get_thread_name()
-    #       self.info.code = self.code()
-    #       self.info.name = type(self).__name__
+    def __init__(self, *args, **kwargs):
+        class_name = type(self).__name__
+        msg_cls = getattr(types_pb2, class_name)
+        if class_name == "Formatting" and len(args) > 0:
+            kwargs["msg"] = args[0]
+            args = ()
+        assert (
+            len(args) == 0
+        ), f"[{class_name}] Don't use positional arguments when constructing logging events"
+        if "base_msg" in kwargs:
+            kwargs["base_msg"] = str(kwargs["base_msg"])
+        if "msg" in kwargs:
+            kwargs["msg"] = str(kwargs["msg"])
+        try:
+            self.pb_msg = ParseDict(kwargs, msg_cls())
+        except Exception as exc:
+            raise Exception(f"[{class_name}]: Unable to parse dict {kwargs},\n exc: {exc}")
```

Comment:
As discussed during the live walkthrough, we want to stop throwing an exception (and stopping the run), and instead just log a warning that we failed to serialize the event:
- https://github.com/dbt-labs/dbt-core/issues/7113

@gshank made the point that it's a bit weird to fire an event _within the code_ for firing events. I don't have strong opinions here on the implementation — just that, as an end user, I would want dbt to more gracefully handle serialization errors.

Reply:
The nested fire_event seems to be working okay. Added a test for it.